### PR TITLE
Try using an enum instead of a table for authorities

### PR DESCRIPTION
**Pull request by jon-betts**
Maybe for:

 * https://github.com/hypothesis/report/issues/4

This is because constantly joining onto a table is a pain. It also means we have to jump through more hoops to make Metabase show this as a multi-select.

This will require a create from scratch to change the data type.

### CODE REVIEW

**Comment by jon-betts** [comment]
Ok, so this is kind of better, and also kind of sucks. 

Using an enum is nice because it appears to be a string field, but has a limited, ordered set of values and stores as an int. This makes all queries against it much simpler, as you just select what you want using simple strings without a join onto the table.

## Issues with recreation

Updating it is an issue. As we need the authorites to define the annotation table it becomes part of the slow process if it requires regenerating the annotation table. Therefore a `DROP ... CASCADE` isn't really an option. 

You can `ALTER` an enum, and we can define the entire enum using alter statements, but Postgres will refuse to use an enum that is _redefined_ during a transaction. So you can define one and use it, but not redefine it and use it in one go. Postgres is smart enough to know that the first time round it's being created, so you don't see an issue during create, but you do when trying the "fast recreate".

This means you have to do it in two transactions, which is a bit startling. This is easily done by dropping a "COMMIT" statement, but it seems a bit weird to have this one aspect of it be non-transactional. As this only applies to the destructive actions, I think this would actually be acceptable, but it doesn't feel as safe.

## Issues with use in other projects

In LMS we don't need to replicate the authority data, just query on it, but if you reference a table with a type you don't have in FDW you can't even look at it without importing the type. This is a case of copy pasting the enum to your local system which we've done in other places, but it's a bit of a shame.

A similar situation is true in the report DB, but we're more likely to need / want the entire enum there to tip off Metabase about it, _however_ you can achieve much the same thing by explicitly telling Metabase about the authorities table. This is more of a pain as you have to remember to do it everywhere it shows up, but it's still possible.

## Helping with imports

Importing Enums kind of sucks because you can't pull them in automatically over FDW like you can views and tables. You _can_ however import the table `pg_enum` and then read the values out of there.

Therefore it's entirely possible to create an SQL query which returns the correct SQL query / queries to create all the enums from a different database. This is kind of meta though, and we'd either need to:

 * Do this manually from time to time and out the output in our scripts
 * Implement so extra step / process which could handle queries that generate queries to execute

## In summary

* Using a table
  * Can be maintained in transaction in all circumstances
  * Can be imported over FDW
  * Fails gracefully if a value doesn't match (this isn't necessarily good)
  * Makes for more complicated queries (one more join)
  * Needs to be manually configured

 * Using an enum
   * Requires two transactions on redefinition (fine if you want to DROP)
   * Must be manually recreated to use with FDW
   * Will fail obviously if there's a value that doesn't match (this is probably good)
   * Leads to simpler more intuitive queries
   * Tips off Metabase about the values of the column

In most cases where we aren't tangled up with the annotation table, enums probably win this battle (just), but in this specific case, I think a table might still be best.
